### ISSUE TITLE
`ShadowedException` cop cause `warning: constant ::TimeoutError is deprecated`

### ISSUE THREAD

**Issue by ShockwaveNN** [rubocop-hq/rubocop#6675]
I'm not 100% sure this is an issue, but I don't think this is correct behaviour

## Steps to reproduce the problem

1. Create `test.rb` file with
```
def foo
  puts 'bar'
rescue TimeoutError
  puts 'foo'
end

```
2. run `rubocop test.rb`

--------

## Expected behavior

Some cop should warn me about deprecation of TimeoutError

## Actual behavior

```
Inspecting 1 file
/home/lobashov/.rvm/gems/ruby-2.5.3/gems/rubocop-0.62.0/lib/rubocop/cop/lint/shadowed_exception.rb:120: warning: constant ::TimeoutError is deprecated
.

1 file inspected, no offenses detected
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.62.0 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-linux)
```

**Comment by pocke** [comment]
Thank you for reporting issue!


https://github.com/rubocop-hq/rubocop/blob/7abc4badefca71f5fd9cb3f9b0e8e0f78801a51e/lib/rubocop/cop/lint/shadowed_exception.rb#L115
This cop evaluates constant name, and `TimeoutError` warns deprecation of itself when it's evaluated.

```ruby
require 'timeout'
p TimeoutError # => warning: constant ::TimeoutError is deprecated
```


I'm not sure we avoid this warning, If we supress the warning, we need a comment why it is supressed.

**Comment by pocke** [comment]
By the way, we can add a cop like `Lint/DeprecatedTimeoutError` for more descriptive warning message.
It should says "TimeoutError is deprecated. Use Timeout::Error instead.".


I think it is the best way if MRI's warning includes "Use Timeout::Error instead.".
But it looks difficult. It uses `deprecate_constant` method to add the deprecation, but it does not have any interface to set "instead".
https://github.com/ruby/ruby/blob/d7976d145193379d8c43f33e9a5714292a40d994/lib/timeout.rb#L125-L130

**Comment by ShockwaveNN** [comment]
@pocke Yeah, I fought that adding cop about this exact message is best solution for this situation

**Comment by rrosenblum** [comment]
It really is a shame that Ruby's message only gives the deprecation and does not supply the replacement. 

> I'm not sure we avoid this warning, If we supress the warning, we need a comment why it is supressed.

```ruby
silence_warnings do
  # Avoid printing deprecation warnings about constants
  converted << Kernel.const_get(exception)
end
```

This would prevent the warning from showing up when RuboCop runs. I think it would make sense for us to silence the warning since the deprecation warning doesn't have anything to do with the RuboCop code itself.

**Comment by Drenmi** [comment]
> It really is a shame that Ruby's message only gives the deprecation and does not supply the replacement.

Indeed. Found another case of this recently. 🙁 

> This would prevent the warning from showing up when RuboCop runs. I think it would make sense for us to silence the warning since the deprecation warning doesn't have anything to do with the RuboCop code itself.

I agree.

**Comment by elmasantos** [comment]
Hey, I just opened a PR with the changes required in order to avoid showing deprecation warnings about  constants.

>I think it would make sense for us to silence the warning since the deprecation warning doesn't have anything to do with the RuboCop code itself.

I agree with this, and I'm not sure if the feature described by @pocke is really needed, but if it is, maybe should be done in a differente branch.
